wrappers/services.go

```diff
@@ -309,6 +309,12 @@ func StopServices(apps []*snap.AppInfo, reason snap.ServiceStopReason, inter int
 		if err := stopService(sysd, app, inter); err != nil {
 			return err
 		}
+		// ensure the service is really stopped on remove
+		if reason == snap.StopReasonRemove && app.RefreshMode.KillMode() == "process" {
```

Comment:
John argued that we need to do this also on snap stop ?

Reply:
Maybe, I think now that this named "stop-mode" we should honor that and only do a hard stop on remove of the snap.